### PR TITLE
Show context of pseudocodes instead whole of them

### PR DESCRIPTION
**Pull request by NyaMisty**

### CODE REVIEW

File: ida_script.py [hunk 1]
```diff
@@ -115,6 +115,21 @@ def decompile(addr):
...
+    if cfunc is None:
+        return None
+    item = cfunc.body.find_closest_addr(addr)
+    y_holder = idaapi.int_pointer()
+    if not cfunc.find_item_coords(item, None, y_holder):
+        return cfunc
+    y = y_holder.value()
+    lines = cfunc.get_pseudocode()
+    retlines = [idaapi.tag_remove(lines[lnnum].line) for lnnum
+                    in range(max(0, y - context_lines),min(len(lines), y + context_lines))]
```
Comment (disconnect3d) [inline ida_script.py]: Can we change this list comprehension to a generator expression (replace `[]` with `()`).

This way we won't create a list just to join it later on :)

Comment (NyaMisty) [reply ida_script.py]: Oh sorry I'm not that familiar with the generator expression (Sorry for my ignorance :( )
Could you please help me change it or give me more details?

**Comment by disconnect3d** [comment]
Sure, just put this into () instead of []
(I for I in ...) Will create a generator while [I for I in ...] Creates a
list

pt., 3 maj 2019, 1:51 PM użytkownik NyaMisty <notifications@github.com>
napisał:

File: ida_script.py (hunk 1 above)
Comment (disconnect3d) [reply ida_script.py]: Sure, just change `[]` to `()`.

```
In [1]: [i for i in [1, 2, 3]]
Out[1]: [1, 2, 3]

In [2]: type([i for i in [1, 2, 3]])
Out[2]: list

In [3]: (i for i in [1, 2, 3])
Out[3]: <generator object <genexpr> at 0x105694a98>

In [4]: x = (i for i in [1, 2, 3])

In [5]: for i in x: print(i)
1
2
3
```

with a generator you don't create a list in memory and just do the computation of next element each iteration instead

**Comment by NyaMisty** [comment]
Oh that's magical! I've pushed a new commit

File: ida_script.py (hunk 1 above)
Comment (NyaMisty) [reply ida_script.py]: Wow that's quite useful. Thanks for pointing that out for me :D

**Pull request merged**